src/ripple/app/tests/JTx.cpp

```diff
@@ -0,0 +1,400 @@
+            ss.add32 (HashPrefix::txMultiSign);
+            st->addWithoutSigningFields(ss);
+            ss.add160(signingForID);
+            ss.add160(e.id());
+            jo["MultiSignature"] = strHex(make_Slice(
+                e.sk().accountPrivateSign(ss.getData())));
+
+        }
+    };
+}
+
+msig2_t::msig2_t (std::vector<std::pair<
+    Account, Account>> sigs)
+{
+    for (auto& sig : sigs)
+    {
+        auto result = sigs_.emplace(
+            std::piecewise_construct,
+                std::make_tuple(std::move(sig.first)),
+                    std::make_tuple());
+        result.first->second.emplace(
+            std::move(sig.second));
+    }
+}
+
+void
+msig2_t::operator()(Env const& env, JTx& tx) const
+{
+    // VFALCO Inefficient pre-C++14
+    auto sigs = std::move(sigs_);
+    tx.signer = [sigs, &env](Env&, JTx& jt)
+    {
+        jt["SigningPubKey"] = "";
+        boost::optional<STObject> st;
+        try
+        {
+            st = parse(jt.jv);
+        }
+        catch(parse_error const&)
+        {
+            env.test.log << pretty(jt.jv);
+            throw;
+        }
+        auto& ja = jt["MultiSigners"];
+        ja.resize(sigs.size());
+        for (auto i = std::make_pair(0, sigs.begin());
+            i.first < sigs.size(); ++i.first, ++i.second)
+        {
+            auto const& sign_for = i.second->first;
+            auto const& list = i.second->second;
+            auto& ji = ja[i.first]["SigningFor"];
+            ji[jss::Account] = sign_for.human();
+            auto& js = ji["SigningAccounts"];
+            js.resize(list.size());
+            for (auto j = std::make_pair(0, list.begin());
+                j.first < list.size(); ++j.first, ++j.second)
+            {
+                auto& jj = js[j.first]["SigningAccount"];
+                jj[jss::Account] = j.second->human();
+                jj[jss::SigningPubKey] = strHex(make_Slice(
+                    j.second->pk().getAccountPublic()));
+
+                Serializer ss;
+                ss.add32 (HashPrefix::txMultiSign);
+                st->addWithoutSigningFields(ss);
+                ss.add160(sign_for.id());
+                ss.add160(j.second->id());
+                jj["MultiSignature"] = strHex(make_Slice(
+                    j.second->sk().accountPrivateSign(
+                        ss.getData())));
+            }
+        }
+    };
+}
+
+void
+seq::operator()(Env const&, JTx& tx) const
+{
+    if (b_)
+        tx.fill_fee = true;
+    else if(! b_)
+        tx.fill_fee = false;
+    else if (boost::indeterminate(b_))
+        tx[jss::Sequence] = v_;
+}
+
+void
+sig::operator()(Env const&, JTx& tx) const
+{
+    if (b_)
+    {
+        tx.fill_sig = true;
+    }
+    else if(! b_)
+    {
+        tx.fill_sig = false;
+    }
+    else if(boost::indeterminate(b_))
+    {
+        // VFALCO Inefficient pre-C++14
+        auto const account = std::move(account_);
+        tx.signer = [account](Env&, JTx& jt)
+        {
```

Comment:
This creates a new account and then captures that by value. You'd be better off just capturing `account_` by value.